app/src/main/java/com/babylon/wallet/android/data/dapp/model/ConnectorExtensionExchangeInteraction.kt

```diff
@@ -0,0 +1,17 @@
+package com.babylon.wallet.android.data.dapp.model
+
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+
+@Serializable
+sealed class ConnectorExtensionExchangeInteraction(
```

Comment:
isn't it enough to put `@JsonClassDiscriminator("discriminator")` on a sealed class instead of declaring discriminator field?

Reply:
From my understanding, it should have been working, but it's not. From my tests, it works only for nested objects.

Reply:
Thanks for your help @jakub-rdx! It works after indicating the type explicitly at encoding.